### PR TITLE
Fixes an issue with getOwnPropertyDescriptor

### LINKED ISSUE #1285: "all native function values should be intrinsics" - what does it mean?
I am trying to run prepack on an existing code, I get this error

```
all native function values should be intrinsics
Invariant Violation: all native function values should be intrinsics
```

It doesn't show me where exactly is the error in the code, so I am asking here. What does it mean, and how can I find where the error comes from?

### PR DESCRIPTION
**Pull request by trueadm**
Release notes: fixes an issue with getOwnPropertyDescriptor when dealing with a native prototype function

This PR fixes #1285, specifically when using `Object.getOwnPropertyDescriptor` on a native prototype function, we have to resort to returning a temporal abstract as this function isn't available at build time. I also renamed all the tests with typos in them.

cc @anilanar

### CODE REVIEW

File: src/intrinsics/ecma262/Object.js [hunk 1]
```diff
@@ -265,8 +266,41 @@ export default function(realm: Realm): NativeFunctionValue {
...
+      realm.useAbstractInterpretation
+    ) {
+      invariant(P instanceof Value);
+      // this will create a property descriptor at runtime
+      let result = AbstractValue.createTemporalFromBuildFunction(
+        realm,
+        ObjectValue,
+        [getOwnPropertyDescriptor, obj, P],
+        ([methodNode, objNode, keyNode]) => t.callExpression(methodNode, [objNode, keyNode])
+      );
```
Comment (anilanar) [inline src/intrinsics/ecma262/Object.js]: What's the difference between `createFromBuildFunction` and its temporal variant?

Comment (trueadm) [reply src/intrinsics/ecma262/Object.js]: We need to create a "temporal" entry in the generator here.

**Comment by anilanar** [comment]
I was confused about the exact definition of intrinsic. By definition, are native accessor properties intrinsic?

**Comment by trueadm** [comment]
@anilanar Many of them are as their contents are not something we want to serialize (as it's all internal implementation details).

**Comment by anilanar** [comment]
@trueadm If so, wouldn't it be nice if `getterFunc.isIntrinsic()` returned true? It currently returns `false` because its `intrinsicName` is undefined.

**Comment by trueadm** [comment]
@anilanar This function might have been missing an intrinsic value intentionally, thus why it was set as `undefined`. I tested giving it an intrinsic value but that didn't fix the core issue here (you're not meant to serialize `Map.prototype.size` as a prototype function, it's strictly an internal getter I believe).

**Comment by anilanar** [comment]
I see. 
Thanks for your code and input, I'll look into other bugs 😃

File: src/intrinsics/ecma262/Object.js (hunk 1 above)
Comment (hermanventer) [reply src/intrinsics/ecma262/Object.js]: A temporal value is something that needs to get created/computed at a particular point in the time line and may have further mutations that occur later on in the time line. An a-temporal value can get computed at any point and does not change.

**Pull request closed without merging**